### PR TITLE
Ethtool update for legacy OS - part 2

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
- Update statistics and ksetting with the legacy OS restriction

### CODE REVIEW

**Comment by juhlee-microsoft** [comment]
```
[Canonical UbuntuServer 16.04-LTS latest] Test Run On           : 04/15/2020 23:26:08
[Canonical UbuntuServer 16.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
[Canonical UbuntuServer 16.04-LTS latest] Test Priority         : 0,1,2,3
[Canonical UbuntuServer 16.04-LTS latest] Initial Kernel Version: 4.15.0-1075-azure
[Canonical UbuntuServer 16.04-LTS latest] Final Kernel Version  : 4.15.0-1075-azure
[Canonical UbuntuServer 16.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 16.04-LTS latest] Total Time (dd:hh:mm) : 0:0:2
[Canonical UbuntuServer 16.04-LTS latest] 
[Canonical UbuntuServer 16.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 16.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 16.04-LTS latest]     1 NETWORK              ETHTOOL-GET-SET-LINK-KSETTING                                                     PASS                  0.4 

[Canonical UbuntuServer 14.04.0-LTS latest] Test Run On           : 04/15/2020 23:26:01
[Canonical UbuntuServer 14.04.0-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 14.04.0-LTS : latest
[Canonical UbuntuServer 14.04.0-LTS latest] Test Priority         : 0,1,2,3
[Canonical UbuntuServer 14.04.0-LTS latest] Initial Kernel Version: 4.4.0-148-generic
[Canonical UbuntuServer 14.04.0-LTS latest] Final Kernel Version  : 4.4.0-148-generic
[Canonical UbuntuServer 14.04.0-LTS latest] Total Test Cases      : 1 (0 Passed, 0 Failed, 0 Aborted, 1 Skipped)
[Canonical UbuntuServer 14.04.0-LTS latest] Total Time (dd:hh:mm) : 0:0:3
[Canonical UbuntuServer 14.04.0-LTS latest] 
[Canonical UbuntuServer 14.04.0-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 14.04.0-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 14.04.0-LTS latest]     1 NETWORK              ETHTOOL-GET-SET-LINK-KSETTING                                                  SKIPPED                 0.39

[Canonical UbuntuServer 12.04.5-LTS latest] Test Run On           : 04/15/2020 23:25:57
[Canonical UbuntuServer 12.04.5-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 12.04.5-LTS : latest
[Canonical UbuntuServer 12.04.5-LTS latest] Test Priority         : 0,1,2,3
[Canonical UbuntuServer 12.04.5-LTS latest] Initial Kernel Version: 3.13.0-117-generic
[Canonical UbuntuServer 12.04.5-LTS latest] Final Kernel Version  : 3.13.0-117-generic
[Canonical UbuntuServer 12.04.5-LTS latest] Total Test Cases      : 1 (0 Passed, 0 Failed, 0 Aborted, 1 Skipped)
[Canonical UbuntuServer 12.04.5-LTS latest] Total Time (dd:hh:mm) : 0:0:3
[Canonical UbuntuServer 12.04.5-LTS latest] 
[Canonical UbuntuServer 12.04.5-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 12.04.5-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 12.04.5-LTS latest]     1 NETWORK              ETHTOOL-GET-SET-LINK-KSETTING                                                  SKIPPED                 0.71 

[Canonical UbuntuServer 14.04.0-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 14.04.0-LTS latest] Test Run On           : 04/16/2020 01:04:56
[Canonical UbuntuServer 14.04.0-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 14.04.0-LTS : latest
[Canonical UbuntuServer 14.04.0-LTS latest] Test Priority         : 0,1,2,3
[Canonical UbuntuServer 14.04.0-LTS latest] Initial Kernel Version: 4.4.0-148-generic
[Canonical UbuntuServer 14.04.0-LTS latest] Final Kernel Version  : 4.4.0-148-generic
[Canonical UbuntuServer 14.04.0-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 14.04.0-LTS latest] Total Time (dd:hh:mm) : 0:0:2
[Canonical UbuntuServer 14.04.0-LTS latest] 
[Canonical UbuntuServer 14.04.0-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 14.04.0-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 14.04.0-LTS latest]     1 NETWORK              ETHTOOL-GET-STATISTICS                                                            PASS                 0.42 

[Canonical UbuntuServer 12.04.5-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 12.04.5-LTS latest] Test Run On           : 04/16/2020 00:52:46
[Canonical UbuntuServer 12.04.5-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 12.04.5-LTS : latest
[Canonical UbuntuServer 12.04.5-LTS latest] Test Priority         : 0,1,2,3
[Canonical UbuntuServer 12.04.5-LTS latest] Initial Kernel Version: 3.13.0-117-generic
[Canonical UbuntuServer 12.04.5-LTS latest] Final Kernel Version  : 3.13.0-117-generic
[Canonical UbuntuServer 12.04.5-LTS latest] Total Test Cases      : 1 (0 Passed, 0 Failed, 0 Aborted, 1 Skipped)
[Canonical UbuntuServer 12.04.5-LTS latest] Total Time (dd:hh:mm) : 0:0:3
[Canonical UbuntuServer 12.04.5-LTS latest] 
[Canonical UbuntuServer 12.04.5-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 12.04.5-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 12.04.5-LTS latest]     1 NETWORK              ETHTOOL-GET-STATISTICS                                                         SKIPPED                 1.07 
```

**Comment by juhlee-microsoft** [comment]
Let's wait for Joe's reply regarding legacy OS new kernel release model.

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh [hunk 1]
```diff
@@ -36,21 +36,31 @@ ChangeMTU $net_interface 4096
...
     LogMsg "$stats" >> ~/summary.log
 fi
-LogMsg "Getting NIC statistics per CPU with ethtool" >> ~/summary.log
-statspcup=$(ethtool -S $net_interface | grep 'queue_')
-if [ $? -ne 0 ]; then
-    LogErr "Failed to get NIC statistics per CPU with ethtool !" >> ~/summary.log
-    SetTestStateFailed
-    exit 0
+if [[ $DISTRO != "ubuntu_x"* ]]; then
+    LogErr "This distro $DISTRO does not have NIC statistics per CPU feature"
```
Comment (LiliDeng) [inline Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh:53]: still that concern, it only support ubuntu? I think it is kernel specified, not distro, we should run against centos, redhat and sles.

**Comment by juhlee-microsoft** [comment]
Design was not approved. I will revert the previous PR also.

**Pull request closed without merging**